c/src/ml-api-remote-service.c

```diff
@@ -154,6 +155,120 @@ _mlrs_get_service_type (gchar * service_str)
   return service_type;
 }
 
+/**
+ * @brief Get ml remote service activation type.
+ */
+static gboolean
+_mlrs_parse_activate (gchar * activate)
+{
+  gboolean ret = TRUE;
+
+  if (activate && g_ascii_strcasecmp (activate, "false") == 0) {
+    ret = FALSE;
+  }
+
+  return ret;
+}
+
+/**
+ * @brief Callback function for receving data using curl.
+ */
+static size_t
+curl_mem_write_cb (void *data, size_t size, size_t nmemb, void *clientp)
+{
+  size_t recv_size = size * nmemb;
+  GByteArray *array = (GByteArray *) clientp;
+
+  if (!array)
+    return 0;
+
+  g_byte_array_append (array, data, recv_size);
```

Comment:
Is data a valid value?
Is there no case where ```nmemb``` is 0?

Reply:
I changed it to check the receiving size. Thanks.